pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-darwin.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv, lib, fetchurl, _7zz }:
+stdenv.mkDerivation rec {
+  pname = "signal-desktop";
+  version = "7.28.0";
+
+  src = fetchurl {
+    url = "https://updates.signal.org/desktop/${pname}-mac-universal-${version}.dmg";
+    hash = "sha256-yPObHZPCXRycuf9VexooWkXDJLReEiTuFSmCxlHA/OU=";
+  };
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ _7zz ];
+
+  dontConfigure = true;
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/Applications
+    cp -r Signal.app $out/Applications
+
+    runHook postInstall
+  '';
+
+  passthru = {
+    updateScript.command = [ ./update.sh ];
+  };
+
+  meta = {
```

Comment:
Should we add a `meta.mainProgram` for this platform?

Reply:
It doesn't seem to work if you make a symlink, some sort of GPU driver issue